packages/service-library/src/servicelib/long_running_tasks/task.py

```diff
@@ -345,7 +346,28 @@ async def _tasks_monitor(self) -> None:
                                 },
                             ),
                         )
-                    result_field = ResultField(str_error=dumps(e))
+                    try:
+                        result_field = ResultField(str_error=dumps(e))
+                    except (
+                        Exception  # pylint:disable=broad-except
```

Comment:
dumps is your own code. maybe you can catch it there and raise a useful exception instead of going broad?

Reply:
![alt text](image.png)

Reply:
Some reasons to why the generic `Exception` is trapped:
- pickle could possibly be raise other errors than the ones declared by it's interface since the user can provide custom pickle dumps implementations for their classes (low risk of this)
- if an error is raised during the entire serialization of the object to a string, it will still be able to provide a result, as it was for now it would just hang and not provide a result, resulting in a timeout.

If there are better ideas on how to deal with these I'll gladly consider them, Thanks!